org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/cleanup/CleanUpUtils.java

```
@@ -0,0 +1,102 @@
+
+/**
+ * Functions for working with JDT ICleanUpCore and ISimpleCleanUp.
+ *
+ * @author datho7561
+ */
+public class CleanUpUtils {
+
+	private CleanUpUtils() {
+	}
+
+	/**
+	 * Returns the clean up context for the given text document.
+	 *
+	 * @param textDocumentId
+	 *            the text document to get the clean up context for
+	 * @param compilerOpts
+	 *            the compiler options to use for the AST
+	 * @param monitor
+	 *            the progress monitor
+	 * @return the clean up context for the given text document
+	 */
+	public static CleanUpContextCore getCleanUpContext(TextDocumentIdentifier textDocumentId, Map<String, String> compilerOpts, IProgressMonitor monitor) {
+		ICompilationUnit unit = JDTUtils.resolveCompilationUnit(textDocumentId.getUri());
+		CompilationUnit ast = createASTWithOpts(unit, compilerOpts, monitor);
+		return new CleanUpContextCore(unit, ast);
+	}
+
+	/**
+	 * Returns a non-null list of text edits for the given clean up.
+	 *
+	 * @param cleanUp
+	 *            the clean up to get the edits for
+	 * @param context
+	 *            the context to perform the clean up on
+	 * @param monitor
+	 *            the progress monitor
+	 * @return a non-null list of text edits for the given clean up
+	 */
+	public static List<TextEdit> getTextEditFromCleanUp(ISimpleCleanUp cleanUp, CleanUpContextCore context, IProgressMonitor monitor) {
+
+		try {
+			ICleanUpFixCore fix = cleanUp.createFix(context);
+			if (fix == null) {
+				return Collections.emptyList();
+			}
+			CompilationUnitChange cleanUpChange = fix.createChange(monitor);
+			org.eclipse.text.edits.TextEdit jdtEdit = cleanUpChange.getEdit();
+			if (jdtEdit == null) {
+				return Collections.emptyList();
+			}
+			TextEditConverter converter = new TextEditConverter(context.getCompilationUnit(), jdtEdit);
+			List<TextEdit> lspEdit = converter.convert();
+			return lspEdit != null ? lspEdit : Collections.emptyList();
+		} catch (CoreException e) {
+			// TODO: log properly
```

Can we log here or do you see some reason to fail silently ? I would think even a basic message like `String.format("Failed to create text edit for %s", cleanUp.getIdentifier())` would be fine.

---

I guess this could get verbose if the user activates saving on particular intervals ?

---

Ooops one second I'll switch over to String.format

---

Okay, should be good. I'm just logging which clean up failed like you suggested.